tests/test_pay.py

```diff
@@ -2630,3 +2630,15 @@ def test_partial_payment(node_factory, bitcoind, executor):
     with pytest.raises(RpcError, match=r'WIRE_FINAL_INCORRECT_HTLC_AMOUNT'):
         l1.rpc.call('waitsendpay', {'payment_hash': inv['payment_hash'],
                                     'parallel_id': 2})
+
+    line = l4.daemon.wait_for_log('print_htlc_onion.py: Got onion')
+    assert "'type': 'tlv'" in line
+    assert "'forward_amount': '500msat'" in line
+    assert "'total_msat': '1000msat'" in line
+    assert "'payment_secret': '{}'".format(paysecret) in line
+
+    line = l4.daemon.wait_for_log('print_htlc_onion.py: Got onion')
+    assert "'type': 'tlv'" in line
+    assert "'forward_amount': '500msat'" in line
+    assert "'total_msat': '1000msat'" in line
+    assert "'payment_secret': '{}'".format(paysecret) in line
```

Comment:
Maybe use a loop to emphasize that these are identical and the order is not important.